### PR TITLE
[1.0] refactor node structure to hide internal fields

### PR DESCRIPTION
**Pull request by KyleAMathews**

### CODE REVIEW

File: packages/gatsby/src/joi-schemas/joi.js [hunk 1]
```diff
@@ -24,12 +24,14 @@ export const pageSchema = Joi.object()
...
-    content: Joi.string(),
-    type: Joi.string().required(),
-    pluginName: Joi.string().required(),
+    internal: Joi.object().keys({
+      contentDigest: Joi.string().required(),
+      mediaType: Joi.string().required(),
+      type: Joi.string().required(),
+      pluginName: Joi.string().required(),
+      content: Joi.string(),
+    }),
```
Comment (jquense) [inline packages/gatsby/src/joi-schemas/joi.js]: required?

Comment (KyleAMathews) [inline packages/gatsby/src/joi-schemas/joi.js]: It's added automatically by api-runner-node

File: packages/gatsby-source-hacker-news/src/gatsby-node.js [hunk 2]
```diff
@@ -103,13 +103,15 @@ fragment commentsFragment on HackerNewsItem {
 
     const storyNode = {
       ...kidLessStory,
-      domain,
-      order: i + 1,
-      parent: `__SOURCE__`,
-      type: `HNStory`,
       children: [...kids.kids.map(k => k.id)],
```
Comment (0x80) [inline packages/gatsby-source-hacker-news/src/gatsby-node.js]: Not part of the PR but I just happened to noticed the redundant `...` here.

File: packages/gatsby-source-hacker-news/src/gatsby-node.js [hunk 3]
```diff
@@ -133,11 +135,13 @@ fragment commentsFragment on HackerNewsItem {
         }
         let commentNode = {
           ..._.omit(comment, `kids`),
-          order: i + 1,
-          type: `HNComment`,
-          parent,
           children: [...comment.kids.map(k => k.id)],
```
Comment (0x80) [inline packages/gatsby-source-hacker-news/src/gatsby-node.js]: Same comment here

File: packages/gatsby/src/joi-schemas/joi.js (hunk 1 above)
Comment (0x80) [inline packages/gatsby/src/joi-schemas/joi.js]: If it's not required according to this validation, shouldn't `contentDigest` then not be required too? Or are those not added in the same way?

Comment (KyleAMathews) [inline packages/gatsby/src/joi-schemas/joi.js]: @jquense to expand on my answer I typed on my phone as my plane took off :-) If we know which plugin created every node we can do optimizations eventually like invalidating only nodes created by a plugin when it updates, etc.

Comment (KyleAMathews) [inline packages/gatsby/src/joi-schemas/joi.js]: @0x80 what's "it" here?

File: packages/gatsby-source-hacker-news/src/gatsby-node.js (hunk 2 above)
Comment (KyleAMathews) [inline packages/gatsby-source-hacker-news/src/gatsby-node.js]: Oh good point :-)
